### PR TITLE
[test] Fix dpkg lock-frontend error when installing python

### PR DESCRIPTION
**Pull request by saimidu**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Fix the `dpkg lock-frontend` error that sometimes arises when running `apt-get update` immediately after launching the EC2 instance

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/ec2.py [hunk 1]
```diff
@@ -911,7 +911,42 @@ def post_process_mxnet_ec2_performance(connection, log_location):
         raise ValueError("total: {}; n: {} -- something went wrong".format(total, n))
 
 
-@retry(reraise=True, wait=wait_fixed(60), stop=stop_after_attempt(5))
```
Comment (kace) [inline test/test_utils/ec2.py:914]: Why drop the retries?

File: test/test_utils/ec2.py [hunk 2]
```diff
@@ -911,7 +911,42 @@ def post_process_mxnet_ec2_performance(connection, log_location):
...
+
+    :param ec2_conn: Fabric SSH connection
+    :return:
+    """
+    apt_daily_services_list = ["apt-daily.service", "apt-daily-upgrade.service", "unattended-upgrades.service"]
+    apt_daily_services = " ".join(apt_daily_services_list)
+    ec2_conn.run(f"sudo systemctl stop {apt_daily_services}")
+    ec2_conn.run(f"sudo systemctl kill --kill-who=all {apt_daily_services}")
+    num_stopped_services = 0
+    # The `systemctl kill` command is expected to take about 1 second. The 60 second loop here exists to force
```
Comment (kace) [inline test/test_utils/ec2.py:928]: nit: Did you want to improve this wait while we're here? Or should we consider this out of scope for this PR?

File: test/test_utils/sagemaker.py [hunk 3]
```diff
@@ -322,7 +289,6 @@ def execute_local_tests(image, pytest_cache_params):
                     f"Image pull for {image} failed.\ndocker images output = {output}"
                 ) from e
         ec2_conn.run(f"tar -xzf {sm_tests_tar_name}")
-        kill_background_processes_and_run_apt_get_update(ec2_conn)
```
Comment (kace) [inline test/test_utils/sagemaker.py:325]: Wanted to double check that we don't see any problems w/ moving this func?

**Review by tejaschumbalkar — APPROVED** [review]

File: test/test_utils/ec2.py (hunk 1 above)
Comment (saimidu) [reply test/test_utils/ec2.py:914]: The `kill` function is a deterministic solution to the problem of the background process preventing us from running `apt-get update` ourselves.

So there is no more need to retry this function in the hope that it will work on the 2nd-5th attempts. It should always work in the first attempt.

File: test/test_utils/ec2.py (hunk 2 above)
Comment (saimidu) [reply test/test_utils/ec2.py:928]: I agree, but I'd rather just merge this in to unblock all the other failures we've been seeing.

File: test/test_utils/sagemaker.py (hunk 3 above)
Comment (saimidu) [reply test/test_utils/sagemaker.py:325]: We're not removing it, as such. We've just changed the order so that it occurs at a different part in the instance setup process, and absorbed it into the `install_python` function because it's a common first step for both EC2 and SM Local tests.

Comment (kace) [reply test/test_utils/sagemaker.py:325]: Ack

**Review by kace — APPROVED** [review]
